src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -7267,24 +7334,45 @@ String StorageReplicatedMergeTree::getSharedDataReplica(
     return best_replica;
 }
 
+
+Strings StorageReplicatedMergeTree::getZeroCopyPartPath(const MergeTreeSettings & settings, DiskType disk_type, const String & table_uuid,
+    const String & part_name, const String & zookeeper_path_old)
+{
+    Strings res;
+
+    String zero_copy = fmt::format("zero_copy_{}", toString(disk_type));
+
+    String new_path = fs::path(settings.remote_fs_zero_copy_zookeeper_path.toString()) / zero_copy / table_uuid / part_name;
+    res.push_back(new_path);
+    if (settings.remote_fs_zero_copy_path_compatible_mode && !zookeeper_path_old.empty())
```

Comment:
Do we really still need this `compatibility` mode?

Reply:
By the way we are using getDefaultSettings here and remote_fs_zero_copy_path_compatible_mode will be always false.

Reply:
We need this for users who already use replication with zero-copy (like Yandex.Cloud) to upgrade version without downtime.
Upgrade process:
- update all replicas one by one with turning on remote_fs_zero_copy_path_compatible_mode
- run python script for convertation existing locks in new variant with cleanup=False
- turn off remote_fs_zero_copy_path_compatible_mode on all nodes
- run python script again with cleanup=True now
For large clusters upgrade process can takes several days, so without compatible there's a non zero chance to lost some shared data, when one replica is already updated, but some locks is not converted yet.